### ISSUE TITLE
Viewer/compiler doesn't work?

### ISSUE THREAD

**Issue by Yeet7** [lervag/vimtex#2578]
### Description

Sorry for the post in here, just didn't know where else to put it. Not sure if this is a VimTeX thing or if it's okular.

So basically title. I'll link my .vimrc, the .tex file, and all errors at the end of the post.

**tldr**; Okular says it has 2 instances running and that it can't open my file. I don't get why. I've tried most of the other viewers and compilers, none of them seem to be working either. All of the compilers I've tried spit out the error VimtexCompile is not a command.

Basically, I've been trying to follow the vimtex installation guide and such, but I haven't had any luck with the viewers. Compiling works beutifully. I type ":VimtexCompile", and within a second it sends back "Compiled Successfully." However, after that, it tries to boot okular, the viewer I chose. Theres two problems after this. Okular first says "There is already a unique Okular Instance running. This instance won't be the unique one." Which makes no sense as there is nothing showing in Task Manager. After that, it says " Could not open file: ..." (Also, just as a side note, after the errors, if I want to reproduce them, I have to restart the entire file.)

**ps**; the file 100% compiles perfectly in LaTeX. Preamble, macros, and letterfonts are all just creating different macros.
also, the Vimtex info file is all hand typed. I went through it but something may be spelled wrong. My copy and paste doesn't work either for some reason. Tried a bunch of things, nothing works. Oh well.

**Pics**;

* [.vimrc](https://gyazo.com/ee6f1bedf4f8360e398d10c9dd1ef137)
* [.tex file](https://gyazo.com/55b96581eb33f9fcaaefaa4c397d5422)
* [okular err. 1](https://gyazo.com/04e6e87ad3c7f45c27e12f695cc99130)
* [okular err. 2](https://gyazo.com/53daa3614de38c220270f9e0245192b9)

### Steps to reproduce

I don't know how to describe the amount of steps I took to get here. Most of my code is jumbled together through many sources while I was trying to fix so many issues. (most were my fault)

### Expected behavior

For okular to compile the file correctly?

### Actual behavior

Two errors. One saying there are multiple existences, the other saying it can't load the file.

### Do you use a latexmkrc file?

No

### VimtexInfo

```yaml
System info:
    OS: Desktop-SM7V63K (Microsoft Windows 10 Home)
    Vim version: VIM 9.0 ()
    Has clientserver: true
    Servername: VIM1

VimTex project: what
    base: what.tex
    root: C:\Users\[REDACTED]\Desktop\Math\Math
    tex: C:\Users\[REDACTED]\Desktop\Math\Math\what.tex
    main parser: current file verifies
    document class: report
    packages: algorithm2e amsfonts amsmath amssymb amsthm array bookmark cancel comment enumitem etoolbox geometry hyperref import mathtools multicol nameref newpxmath pdfpages tcolorbox theoremref tikz tikz-cd tikzsymbols titletoc transparent varwidth xcolor xfrac xifthen
    source files:
        what.tex
        preamble.tex
        macros.tex
        letterfonts.tex
    compiler: arara
        options:
            --log
    viewer: General
    qf method: LaTeX logfile
```

**Comment by lervag** [comment]
> **tldr**; … All of the compilers I've tried spit out the error VimtexCompile is not a command.

Well, this is a serious warning. It sounds like you've not installed VimTeX correctly!

> Basically, I've been trying to follow the vimtex installation guide and such, but I haven't had any luck with the viewers. Compiling works beutifully.

But did you not just write, above, that you get an error on compile?!

> I type ":VimtexCompile", and within a second it sends back "Compiled Successfully."

Ok; perhaps you _did_ install it properly. But the warning about `:VimtexCompile` not being a command should not depend on how you configure your viewer!

> However, after that, it tries to boot okular, the viewer I chose. Theres two problems after this. Okular first says "There is already a unique Okular Instance running. This instance won't be the unique one." Which makes no sense as there is nothing showing in Task Manager. After that, it says " Could not open file: ..." (Also, just as a side note, after the errors, if I want to reproduce them, I have to restart the entire file.)

I'll look into your config after writing this first response.

> **ps**; the file 100% compiles perfectly in LaTeX. Preamble, macros, and letterfonts are all just creating different macros. also, the Vimtex info file is all hand typed. I went through it but something may be spelled wrong. My copy and paste doesn't work either for some reason. Tried a bunch of things, nothing works. Oh well.

> **Pics**;

Please note that

1. You can insert pictures directly into the GitHub issue. Much better than having to click the link!

2. For small code snippets, it is much better to include the code as [code snippets](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks).

> ### Steps to reproduce
> 
> I don't know how to describe the amount of steps I took to get here. Most of my code is jumbled together through many sources while I was trying to fix so many issues. (most were my fault)

I think you may be overthinking what is meant by "steps to reproduce". The idea is that you provide:

1. Minimal configuraton files (e.g. a minimal `_vimrc` file).
2. A list of steps you use to produce the issue or error, e.g. (not necessarily correct in your case!):

  a. Start Vim with the following `_vimrc` (then include contents).
  b. Open LaTeX file (`:e myfile.tex`).
  c. Run `\ll` - observe 1) document is properly compiled and 2) Okular opens after compilation.
  d. Run `\lv`.

> ### Expected behavior
> 
> For okular to compile the file correctly?

Okular is _not_ a compiler. Okular is a _viewer_.

**Comment by lervag** [comment]
I've looked over your config. May I suggest the following rewrite?

```vim
call plug#begin('~\vimfiles\plugged')
Plug 'lervag/vimtex'
Plug 'junegunn/vim-easy-align'
Plug 'tpope/vim-sensible'
call plug#end()

""" Set options
" Numbers each line and allows backspace to work PROPERLY
set nu
set backspace=indent,eol, start

" Sets font and background colours
if has("gui_running")
  set guifont=Lucida_Console:h12:cANSI:qDRAFT
  colorscheme elflord
endif

""" Configure mappings
let maplocalleader = ','

""" Configure plugins
" VimTeX
let g:vimtex_compiler_method = 'arara'
let g:vimtex_view_general_viewer = 'okular'
let g:vimtex_view_general_options = '--unique file:@pdf\#src:@line@tex'
```

First, I would load the plugins at the top. The `call plug#end()` also enables syntax and does `filetype plugin indent on`. Then I would set the regular Vim options. I would configure any global mapping you want and set the maplocalleader, and finally have a section for plugin configuration beneath. Also, you don't need to specify the general viewer, because it is already the default.

Now, You use `arara` as a compiler. Why is that? `latexmk` usually works well for most people. `arara` is neat, but I would say it is more of an expert system. Also, `latexmk` generally has more "bells and whistles" with VimTeX, so, especially for people without much experience, I would strongly recommend it.

**Comment by Yeet7** [comment]
I apologize, I used compiler and viewer interchangeably. I understand they aren't, I just didn't realize. Apologies for the images as well.
The reason I use arara is because when I put latexmk in, it responds with "E492L Not an editor command: VimtexCompile"
Also, for whatever reason, I cannot get the shortcut \ll to work. Windows just screams at me with the default noise.  I don't really understand why this is.
I also tried your code, and it works beautifully with okular. I just get two different errors. One now popping up whenever I open a file, which is literally blank (I'll post a ss the right way :) and the other, which I think is a thing on my end, saying LaTeX font error and that label(s) may have changed.

![6075b17ed53ee33ac82bd6fa1785c3c3](https://user-images.githubusercontent.com/51092112/203868979-46f1b62c-8a55-4040-b360-784647db82d4.png)
![968f9466b452cd54fd4f6bb9929fbc1c](https://user-images.githubusercontent.com/51092112/203868984-4786bee6-1637-476f-bc7c-391744caaa7b.png)

### Steps to reproduce (new and improved)
1. Use my .vimrc and boot up my what.tex file.
2. A blank error should appear. (doesn't seem to have any affect on the code/file itself)
3. Use command :VimtexCompile and wait.
4. Okular should open, no errors should occur.

**Comment by Yeet7** [comment]
Alrighty, it's done. I've figured it out. I have literally zero clue as to why this was happening, so any explanation would help a lot. I've gone through the help pages and haven't seen anything to indicate why this is. 

Basically, I've removed everything relating to VimTeX from my .vimrc. The only things I've kept are the  `let maplocalleader = ','` and the vim-plug line. That's it. Everything now functions beautifully with okular. I type `VimtexCompile` and it compiles, completes, and opens up okular. No errors, and better yet, it autocompiles perfectly.

The only thing that still does not work is the \ll command, which still screams the windows default noise. 

Here's my new .vimrc file that works;
![79053503b59561052e9bc754c4500dc2](https://user-images.githubusercontent.com/51092112/204061077-d159a9ac-5cc0-4b32-a87c-08817d4d551c.png)

**Comment by lervag** [comment]
I'll respond to your last comment first:

> Basically, I've removed everything relating to VimTeX from my .vimrc. The only things I've kept are the `let maplocalleader = ','`
> 
> The only thing that doesn't work now are the \ commands (i.e. \ll and \lk). It just screams the windows default "this isn't a thing" noise.

Well, that's because you've set your leader to `,`. See `:help maplocalleader` and `help vimtex-default-mappings`. Due to this, `\ll` becomes `,ll`.

> Here's my new .vimrc file;

Again, and as an advice for future comments/posts, it is better if you instead include the _text_. It should be easy to copy the contents of your Vim buffer and paste it.

One thing, though: it surprises me that VimTeX opens _Okular_ if you did not configure that. By default, VimTeX would use SumatraPDF on Windows, if it is available. Are you sure this is not the case now? That is, the PDF viewer that is opened by VimTeX, is it really Okular, or is it SumatraPDF?

----

> You may very well be correct that I have installed vimtex incorrectly or that this could all be fixed in a single line. I'm new to vim and latex (if you hadn't noticed already lol) and have been googling things this entire time, squeezing code together trying to make it work. Thanks for the help.

I find the problems interesting and somewhat hard to understand. I'm curious, after you open a LaTeX file, what's the output of `:scriptnames`?

**Comment by Yeet7** [comment]
> Well, that's because you've set your leader to `,`. See `:help maplocalleader` and `help vimtex-default-mappings`. Due to this, `\ll` becomes `,ll`.

I see, however it still doesn't work. I've set it to both, and have tried both, but it still screams the default noise. This isn't as big of an issue for me, however.

> One thing, though: it surprises me that VimTeX opens _Okular_ if you did not configure that. By default, VimTeX would use SumatraPDF on Windows, if it is available. Are you sure this is not the case now? That is, the PDF viewer that is opened by VimTeX, is it really Okular, or is it SumatraPDF?

100% Okular. I have SumatraPDF downloaded as well, but it only opens Okular. It doesn't even recognise `SumatraPDF` when I put it as the reader. I thought it was weird too, but yeah it is 1000% Okular.

> I find the problems interesting and somewhat hard to understand. I'm curious, after you open a LaTeX file, what's the output of `:scriptnames`?

A lot of different files. (sorry, I literally can't copy this for some reason. It just starts spamming new lines in the command (line?). Here are some screen shots.

![63f0bd80116ddcbea4a9c956cd3f6216](https://user-images.githubusercontent.com/51092112/204108286-6c0e3924-c137-4380-8cbe-36cbf7a87664.png)
![9a938964ea7cd6b3defd1cbc016f02b6](https://user-images.githubusercontent.com/51092112/204108289-0d477236-55a1-459b-86cb-a507577787f1.png)
![83c3fe1b51ba1fe52c0c6a5af5874fbf](https://user-images.githubusercontent.com/51092112/204108292-5f6d0145-0333-4c58-808b-8f315f5381ca.png)
![4ab72deaa7658d5b9affdbc7ba775004](https://user-images.githubusercontent.com/51092112/204108294-768d31d8-fe94-4091-a12f-4ba35c530272.png)
![602d0905ea226e0b274771680d158ef3](https://user-images.githubusercontent.com/51092112/204108302-91ca4eef-6b38-4fa8-82af-09402c3f7917.png)

**Comment by lervag** [comment]
> > Well, that's because you've set your leader to `,`. See `:help maplocalleader` and `help vimtex-default-mappings`. Due to this, `\ll` becomes `,ll`.
> 
> I see, however it still doesn't work. I've set it to both, and have tried both, but it still screams the default noise. This isn't as big of an issue for me, however.

Ok; what happens if you just remove all lines where you do anything with `maplocalleader`? The it should use the default, which is `\ll`. Does `\ll` work now?

> > One thing, though: it surprises me that VimTeX opens _Okular_ if you did not configure that. By default, VimTeX would use SumatraPDF on Windows, if it is available. Are you sure this is not the case now? That is, the PDF viewer that is opened by VimTeX, is it really Okular, or is it SumatraPDF?
> 
> 100% Okular. I have SumatraPDF downloaded as well, but it only opens Okular. It doesn't even recognise `SumatraPDF` when I put it as the reader. I thought it was weird too, but yeah it is 1000% Okular.

Ok. VimTeX does not find `SumatraPDF` probably because you did not install it properly - please read `:help vimtex-faq-windows` and note the thing about the `PATH`. If it works properly, then you should be able to start SumatraPDF in a `cmd` prompt. To test, press windows menu and start the `cmd` prompt, then type `SumatraPDF.exe`. If SumatraPDF starts, then it should also work with VimTeX.

What seems to happen here, is that VimTeX uses the `start` utility, which will use Okular because it is the _default_ viewer on your system.

> > I find the problems interesting and somewhat hard to understand. I'm curious, after you open a LaTeX file, what's the output of `:scriptnames`?
> 
> A lot of different files. ...

You are basing your config on [this](https://github.com/amix/vimrc), right? In addition, you are using vim-plug to load additional plugins, such as VimTeX. IMHO, this is not so good, because it becomes a mess to undertand what comes from where. The config from amix overrides your own config (your `.vimrc` file is the first config file sourced). Thus, I would _strongly_ advice that you remove amix from your configuration and instead copy the parts and plugins you want. This will probably take you quite some time, though; be warned. But you will probably also learn a lot.

Notice, by the way, that you've added nerdtree both in your vim-plug config _and_ it is added by the amix/vimrc stuff. That's not good.
